src/main/java/core/basesyntax/SupplyOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.put(transaction.getFruit(), storage.getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());
```

Comment:
The `handle` method should ensure that it does not directly depend on the concrete implementation of the storage. Instead, consider defining an interface for the storage and using it here. This will make your code adhere to the Dependency Inversion principle, one of the SOLID rules.